WordPress/Classes/ViewRelated/Gutenberg/GutenbergViewController.swift

```diff
@@ -648,6 +653,49 @@ extension GutenbergViewController: GutenbergBridgeDelegate {
         mediaInserterHelper.cancelUploadOf(media: media)
     }
 
+    func gutenbergDidRequestToSetFeaturedImage(for mediaID: Int32) {
+        // dismiss current controller in order to allow the new alert controller to be presented
+        presentedViewController?.dismiss(animated: false, completion: nil)
+
+        let featuredImageId = post.featuredImage?.mediaID
+        let mediaIdNoFeaturedImageSet = 0
+
+        if featuredImageId != nil && mediaID == featuredImageId as! Int32 {
+            // nothing special to do, trying to set the image that's already set as featured
+            return
+        }
+
+        if mediaID == mediaIdNoFeaturedImageSet {
+            // user tries to clear the featured image setting
+            featuredImageHelper.setFeaturedImage(mediaID: mediaID)
+            return
+        }
```

Comment:
Again you could consider using the guard-statement. Note that the condition is negated when using it (i.e. only continue if the media ID _is not_ zero, otherwise do some clean up and exit):
```
        guard mediaID != mediaIdNoFeaturedImageSet else {
            // user tries to clear the featured image setting
            featuredImageHelper.setFeaturedImage(mediaID: mediaID)
            return
        }
```